tests/test_lane.py

```diff
@@ -3367,76 +3366,86 @@ def assert_contents(g, expect):
                 # (two) titles each. The 'Science Fiction' lane
-                # features a middle-quality work that was already
-                # featured above in a list, even though there's a
-                # low-quality work that could have been used
-                # instead. Each lane query has its own LIMIT applied,
-                # so we didn't even see the low-quality work.
-                (hq_sf, sf_lane),
-                (mq_sf, sf_lane),
-                (hq_ro, romance_lane),
-                (mq_ro, romance_lane),
+                # features a low-quality work because the
+                # medium-quality work was already used above.
+                (self.hq_sf, sf_lane),
+                (self.lq_sf, sf_lane),
+                (self.hq_ro, romance_lane),
+                (self.mq_ro, romance_lane),
 
                 # The 'Discredited Nonfiction' lane contains a single
                 # book. There just weren't enough matching books to fill
                 # out the lane to FEATURED_LANE_SIZE.
-                (nonfiction, discredited_nonfiction),
+                (self.nonfiction, discredited_nonfiction),
 
                 # The 'Fiction' lane contains a title that fits in the
                 # fiction lane but was not classified under any other
                 # lane. It also contains a title that was previously
-                # featured earlier. There's a low-quality litfic title
-                # in the database, but we didn't see it because the
-                # 'Fiction' query had a LIMIT applied to it.
-                (hq_litfic, fiction),
-                (hq_ro, fiction),
+                # featured earlier. The search index knows about a
+                # title (lq_litfix) that was not previously featured,
+                # but we didn't see it because the Elasticsearch query
+                # didn't happen fetch it. The query for each lane
+                # happens separately and there were too many
+                # high-quality works in 'fiction' for the low-quality
+                # one to show up.
+                (self.hq_litfic, fiction),
+                (self.hq_sf, fiction),
             ]
         )
 
         # If we ask only about 'Fiction', not including its sublanes,
-        # we get the same books associated with that lane.
-        #
-        # hq_ro shows up before hq_litfic because its .random is a
-        # larger number. In the previous example, hq_ro showed up
-        # after hq_litfic because we knew we'd already shown hq_ro in
-        # a previous lane.
+        # we get only the subset of the books previously returned for
+        # 'fiction'.
         assert_contents(
-            fiction.groups(self._db, include_sublanes=False),
-            [(hq_ro, fiction), (hq_litfic, fiction)]
+            make_groups(fiction, include_sublanes=False),
+            [(self.hq_litfic, fiction), (self.hq_sf, fiction)]
```

Comment:
I believe using the deterministic setting makes the results  more consistent here across sub-tests.